SignalServiceKit/src/Util/TypingIndicators.swift

```diff
@@ -238,6 +292,14 @@ public class TypingIndicatorsImpl: NSObject, TypingIndicators {
         private func sendTypingMessage(forThread thread: TSThread, action: TypingIndicatorAction) {
             Logger.verbose("\(TypingIndicatorMessage.string(forTypingIndicatorAction: action))")
 
+            guard let delegate = delegate else {
+                owsFailDebug("Missing delegate.")
+                return
+            }
+            guard delegate.areTypingIndicatorsEnabled() else {
+                return
```

Comment:
`sendTypingMessage` is a bit misleading of a method name since we'll bail out here.

or is this just a safety, in which cases this should have an owsfailDebug?

Reply:
`areTypingIndicatorsEnabled` reflects the user-facing setting in the app preferences.  If it's disabled we don't want to emit these messages or show the indicators for other users.  

I'll rename the method and improve the comments.